src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -680,7 +680,10 @@ private void ReturnStream(Http2Stream stream)
                 stream.DrainExpirationTicks = SystemClock.UtcNowTicks + StreamPoolExpiryTicks;
 
                 StreamPool.Push(stream);
+                return true;
             }
+
+            return false;
```

Comment:
Why not just dispose it here?

Reply:
Having a flag on ReturnStream that indicates whether it was successful or not felt better to me 🤷

Reply:
Feels over engineered 😏. This pool is literally for streams only

Reply:
I also slightly prefer disposing here. If we do, we should just blindly call `ReturnStream` and check `CanReuse` before disposing here too.

Reply:
Yea, lets move both checks inside here.

Reply:
I'm going to remove ReturnStream altogether. It is only used in one place, and RemoveStream would just turn into:
```cs
_streams.Remove(stream);
ReturnStream(stream);
```
Which seems kind of pointless.